### ISSUE TITLE
Possibility to send alert in case of Grafana datasource outage

### ISSUE THREAD

**Issue by dgdevops** [grafana/grafana#36520]
**What would you like to be added**:
A feature that allows to receive alerts in case Grafana experiences error(s) reaching a datasource.

**Why is this needed**:
Due to unexpected network/datasource app issues Grafana can lose connection to its datasources resulting in false alerts and "no data". It would be proactive if Grafana could send out an alert in these events.

**Comment by Elfo404** [comment]
Hi @dgdevops, 

if you are using the new alerting system, you can configure what I think you are describing when creating an alert rule:

<img width="456" alt="Screenshot 2021-07-09 at 11 38 17" src="https://user-images.githubusercontent.com/1170767/125058057-c8506300-e0a1-11eb-9317-c87b2cf4c03c.png">

or in the old alerting system:

<img width="534" alt="Screenshot 2021-07-09 at 11 42 38" src="https://user-images.githubusercontent.com/1170767/125058638-63493d00-e0a2-11eb-9d42-bbf5e95dc394.png">

**Comment by dgdevops** [comment]
Hello @Elfo404 ,
Thank you for your response.
I am not sure why this has been closed, the proposed solution you showed above is just for receiving an alert when there is "No Data" and cannot be used to detect if there is "No Data" because of a datasource issue. Many of our panels are set up to have "No data" by default and we receive alerts when there is actual data (means there is a problem). Is there an option to check actual datasource issues?

**Comment by Elfo404** [comment]
If datasource is not reachable (because of whatever reason), setting `If execution error or timeout` to `Alerting` should trigger alerts. given what you wrote, you can also set `if no data or all values are null` to `Ok` and you should get your desired behavior. does it work for you?
<img width="487" alt="Screenshot 2021-07-12 at 12 42 32" src="https://user-images.githubusercontent.com/1170767/125274845-a3145c80-e30e-11eb-9e4a-8e472d284403.png">

**Comment by dgdevops** [comment]
Hello @Elfo404,
Thank you for your response.
I would say the `If execution error or timeout` would work as a workaround but it would require to set up a query with a random metric for each datasource (in our case its Prometheus). For now I configured the Prometheus instances as blackbox-exporter targets and I am using the HTTP response codes as a base for alerting which is more reliable I believe.

**Comment by zoltanbedi** [comment]
@nathanrodman @peterholmberg what do you think about this?

**Comment by nathanrodman** [comment]
Sounds like a good use case for using Synthetic Monitoring to monitor the status of the datasource. You can create alert rules to then notify you if one goes down https://grafana.com/grafana/plugins/grafana-synthetic-monitoring-app/

**Comment by dgdevops** [comment]
Hello @nathanrodman ,
Thank you for the suggestion.
I looked into the Synthetic Monitoring plugin before opening this request but our Prometheus instances are only available internally so the internal probe locations wouldn't really help here. As I checked there is a way to deploy own monitoring agent(s) but it just adds an extra item to the maintenance portfolio so I was looking for a solution the utilizes already existing tools hence I decided to use our own blackbox-exporter.
